Brokerages/Zerodha/TextFieldParser.cs

```diff
@@ -0,0 +1,1943 @@
+        /// </summary>
+        /// <value>The expression</value>
+        /// <remarks></remarks>
+        private string EndQuotePattern
+        {
+            get
+            {
+                return string.Format(CultureInfo.InvariantCulture, ENDING_QUOTE, WhitespacePattern);
+            }
+        }
+
+        /// **************************************************************
+        /// ;WhitepaceCharacters
+        /// <summary>
+        /// Returns a string containing all the characters which are whitespace for parsing purposes
+        /// </summary>
+        /// <value></value>
+        /// <remarks></remarks>
+        private string WhitespaceCharacters
+        {
+            get
+            {
+                StringBuilder builder = new StringBuilder();
+                foreach (int code in m_WhitespaceCodes)
+                {
+                    char spaceChar = (char)code;
+                    if (!CharacterIsInDelimiter(spaceChar))
+                        builder.Append(spaceChar);
+                }
+
+                return builder.ToString();
+            }
+        }
+
+        /// **************************************************************
+        /// ;WhitespacePattern
+        /// <summary>
+        /// Gets the character set of whitespaces to be used in a regex pattern
+        /// </summary>
+        /// <value></value>
+        /// <remarks></remarks>
+        private string WhitespacePattern
+        {
+            get
+            {
+                StringBuilder builder = new StringBuilder();
+                foreach (int code in m_WhitespaceCodes)
+                {
+                    char spaceChar = (char)code;
+                    if (!CharacterIsInDelimiter(spaceChar))
+                        // Gives us something like \u00A0
+                        builder.Append(@"\u" + code.ToString("X4", CultureInfo.InvariantCulture));
+                }
+
+                return builder.ToString();
+            }
+        }
+
+        /// ************************************************************************************
+        /// ;CharacterIsInDelimiter
+        /// <summary>
+        /// Checks to see if the passed in character is in any of the delimiters
+        /// </summary>
+        /// <param name="testCharacter">The character to look for</param>
+        /// <returns>True if the character is found in a delimiter, otherwise false</returns>
+        /// <remarks></remarks>
+        private bool CharacterIsInDelimiter(char testCharacter)
+        {
+            //Debug.Assert(m_Delimiters != null, "No delimiters set!");
+
+            foreach (string delimiter in m_Delimiters)
+            {
+                if (delimiter.IndexOf(testCharacter) > -1)
+                    return true;
+            }
+
+            return false;
+        }
+
+        // Indicates reader has been disposed
+        private bool m_Disposed;
```

Comment:
Our coding convention is to declare private members at the top of the class

Reply:
Will refactor the code to use CSVHelper library instead and get rid of this class altogether @StefanoRaggi. Meanwhile can we focus on making the implementation functional and then will cleanup everything at once?